### PR TITLE
Create file blocks when dropping multiple files at once

### LINKED ISSUE #11249: Dropping multiple files should create file blocks
**Describe the bug**
In the Classic Editor you can drop multiple files into the editor which opens the media modal to upload them so you can add them in bulk to the content. That's currently not possible with the block editor.

This is similar to #4148 for images and the gallery block.

### PR DESCRIPTION
**Pull request by ocean90**
## Description

I initially checked for the file type to not be one of the other types supported by the video/image/audio/gallery blocks. But since this has a higher priority this isn't actually required as it will only be used as a fallback. Due to the power of the block specific transforms it's still possible to easily convert the block to another specific media block which I think is just awesome.

Fixes #11249.

## Screenshots 
kapture 2018-10-31 at 13 31 16

### CODE REVIEW

File: packages/block-library/src/file/index.js [hunk 1]
```diff
@@ -77,20 +77,27 @@ export const settings = {
 		from: [
 			{
 				type: 'files',
-				isMatch: ( files ) => files.length === 1,
+				isMatch( files ) {
+					return files.length > 0;
```
Comment (ocean90) [inline packages/block-library/src/file/index.js]: `isMatch` is required per https://github.com/WordPress/gutenberg/blob/fe51bd08684462690988de7ec61a3cdda083710a/packages/editor/src/components/block-drop-zone/index.js#L62 so this is just a simple sanity check.

File: packages/block-library/src/file/index.js [hunk 2]
```diff
@@ -77,20 +77,27 @@ export const settings = {
...
-					const blobURL = createBlobURL( file );
+					const blocks = [];
 
-					// File will be uploaded in componentDidMount()
-					return createBlock( 'core/file', {
-						href: blobURL,
-						fileName: file.name,
-						textLinkHref: blobURL,
+					files.map( ( file ) => {
+						const blobURL = createBlobURL( file );
```
Comment (youknowriad) [inline packages/block-library/src/file/index.js]: Out of scope of this PR but I thought I'd mention for context. Ideally, we shouldn't rely on the `edit` function to properly upload the file and use the final URL. Instead we should avoid storing temporary attributes in blocks. 

That said, we don't have a proper solution for this problem at the moment, we'll more likely need to think about a "reactive" transform API but this is not high-priority.

**Review by youknowriad — APPROVED** [review]
LGTM 👍

**Comment by jasmussen** [comment]
♥️

**Pull request merged**